pkg/spanstat/spanstat.go

```diff
@@ -43,19 +45,30 @@ func Start() *SpanStat {
 
 // Start starts a new span
 func (s *SpanStat) Start() *SpanStat {
+	s.mutex.Lock()
```

Comment:
FYI, If you compose the `SpanStat` from `lock.RWMutex` directly (ie no field name), then these can be `s.Lock()` / `defer.sUnlock()` and you don't have to worry about whether to call it `mutex` or `lock` or so on :-)

FWIW we aren't consistently composing objects from locks this way across the codebase so I'm not fussed about which approach we take here.

Reply:
thanks, it's good to know. Composing really looks code look nicer. To make it easier for other reviewers, I will just leave as it is now. Will do it in another PR if required.